lib/utils/checkout.js

```diff
@@ -8,13 +8,24 @@ const getRedirectUrl = (redirectUrl) => {
   return new URL(redirectUrl, window.location.origin).href;
 };
 
-export const getCheckoutPayload = (items) => {
+export const getCheckoutPayload = (items, redirectUrl) => {
   return {
-    input: {
-      lineItems: items.map((item) => ({
+    lines: items.map((item) => {
+      /*To add a shopify subscription to the cart with their
+      checkoutCreate mutation, you must add its selling plan id.
+      To add a product, you must add its variant id.
+      Merchandise id is required in any case.*/
+
+      const line = {
         quantity: item.quantity,
-        variantId: item.variants.edges[0].node.id
-      }))
-    }
+        merchandiseId: item.storefrontData.variants.edges[0].node.id
+      };
+      if (item.purchaseType === 'recurring') {
+        line.sellingPlanId = item.storefrontData.sellingPlanGroups.edges[0].node.sellingPlans.edges[0].node.id;
```

Comment:
I added lodash get here with an empty string default value for now. There is an alternative way to get the product's selling plan ID, but for some reason I can't get it working properly in TakeShape. The fields I need are returning null. So I'm gonna try to get that working.

Otherwise, as I mentioned in another comment, I'm not sure if I'm using lodash optimally here.

Reply:
I added lodash here. Changed up the schema a little to fetch some data that could be a fallback if this fails. Unfortunately I have to use my own `MyShopifyStorefront_Product` shape to do this, because if I use `shopify-storefront:Product` as the shape, it won't index the `storefrontData` deeply enough.

I made a ticket for it:
https://app.shortcut.com/takeshape/story/8564/indexer-seems-to-fetch-data-inconsistently-on-properties-that-use-resolver

Reply:
That's not the issue, it has nothing to do with indexed data in this case. 

If you attempt to access a property of an undefined object you will throw an error. If you have properties that may or may not exist on the path to the property you're trying to access it's generally not safe to to access it via a plain JS path unless you use optional chaining, have a typed guarantee or use a safe access function. 

## Does not throw

```
line.sellingPlanId = get(item, 'storefrontData.sellingPlanGroups.edges[0].node.sellingPlans.edges[0].node.id')
```

## May throw

You have 10 paths on the way to `id`. 10 opportunities to throw.

```
line.sellingPlanId = item.storefrontData.sellingPlanGroups.edges[0].node.sellingPlans.edges[0].node.id;
```